File: DiagonalDifference.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */  
    int n;
    cin>>n;
    int a[n][n];
    int i,j;
    int x=0,y=0;
    for(i = 0 ; i < n ; i++)
        {
        for(j = 0 ; j < n ; j++)
            {
            
        
        cin>>a[i][j];
            
            if(i == j)
                x += a[i][j];
            if(i + j + 1 == n)
                y += a[i][j];
        }
    }
    cout<<abs(x-y);
    return 0;
}

File: DiagonalDifference.go
package main
import (
    "fmt"
)

func main() {
 //Enter your code here. Read input from STDIN. Print output to STDOUT
    var n int
    fmt.Scanf("%v",&n)
    //const x = n
    //var array[x][x] int
    //var number int
    var left_diagonal int
    var right_diagonal int
    /*
    for i:=0;i<n;i++{
        for j:=0;j<n;j++{
            fmt.Scanf("%v",&number)
            
        }
    }
    */
    array := make([][]int,n)
    var number int
    for i:=0;i<n;i++{
        array[i] = make([]int,n)
        for j:=0;j<n;j++{
            fmt.Scanf("%v",&number)
            array[i][j] = number
            if i==j{
                left_diagonal += array[i][j]
            }
            if i+j == n-1{
                right_diagonal += array[i][j]
            }
        }
    }
    
    if left_diagonal-right_diagonal > 0{
        fmt.Println(left_diagonal-right_diagonal)
    }else{
        fmt.Println(right_diagonal-left_diagonal)
    }
    
}
